mountpoint-s3-client/tests/put_object.rs

```diff
@@ -95,3 +98,37 @@ async fn test_put_object_large(client: &impl ObjectClient, bucket: &str, prefix:
 }
 
 object_client_test!(test_put_object_large);
+
+// Test for dropped PUT object. Checks that the GET fails.
+async fn test_put_object_dropped(client: &impl ObjectClient, bucket: &str, prefix: &str) {
```

Comment:
Yes, I've added that to the tests in `mountpoint`, where we already use the SDK client, but not here in `mountpoint-s3-client`. I'll try, but it may require a bit of reshuffling.

Reply:
Actually, in `mountpoint` I'm only testing that the mpu is in progress. We don't exercise the abort there (yet?). 

So I still need to the SDK test dependency to s3-client, but also figure out how to test if the abort happens asynchronously...

Reply:
I've added this to a new test, just for S3 (no mock). It calls `sleep` to wait for the abort, not sure what better options we have...

(deleted a couple of previous comments: we are already using the SDK client in `s3-client` tests)